src/linter/block.go

```diff
@@ -1773,6 +1781,51 @@ func (b *BlockWalker) handleAssign(a *ir.Assign) bool {
 	return false
 }
 
+func (b *BlockWalker) handleAssignOp(o ir.Node) (res bool) {
+	// We are converting "$x += $y" to "$x = $x + $y"
+	// to reuse existing processing code.
+	switch s := o.(type) {
+	case *ir.AssignPlus:
+		as := ir.Assign{
+			Variable: s.Variable,
+			Expression: &ir.PlusExpr{
+				Left:  s.Variable,
+				Right: s.Expression,
+			},
```

Comment:
I think it's better to set positions info here.

Otherwise, when the walker will want to emit a warning it'll give it for some undefined position.